### ISSUE TITLE
SEO URL's by default?

### ISSUE THREAD

**Issue by lucasjkr** [copona/copona#82]
How do we feel about creating an .htaccess file during installation and turning on SEO URL's by default?

Or have it be a selectable option during installation (rather than the two-step process, turning on the setting then renaming htaccess.txt?

**Comment by prhost** [comment]
Need support for at least htaccess from apache and nginx

http://stackoverflow.com/questions/15625928/opencart-on-nginx

**Comment by arnisjuraga** [comment]
Yes, SEO must be compatible with hostings, which does not have `mod_rewrite`, it's like Nginx or Dev sites, where mod_rewrite is not enabled.

I would vote for `.htaccess` on by default at the moment. But Copona should make some checks - if `RewriteEngine` is not on/working, then fall back to non-SEO links?

**Comment by lucasjkr** [comment]
Looks like we'd need to evaluate Nginx and Apache for rewrites separately.

For Apache, in the step_3.php, we could add a check like this:

if(in_array('mod_rewrite', apache_get_modules()) && stristr($string, 'apache') !== FALSE){
   ... create .htaccess file ...
   ... update settings table to turn on seo_urls ...
}

Is that OK? Or would you prefer more or different methods of checking?
